File: app/src/main/java/edgar/yodgorbek/sportnews/model/Article.java
package edgar.yodgorbek.sportnews.model;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.ForeignKey;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;


@Entity(
        tableName = "article",
        foreignKeys = @ForeignKey(
                entity = Source.class,
                parentColumns = "id",
                childColumns = "source_id"
        ))
public class Article {
    @PrimaryKey
    @NonNull
    @SerializedName("source")
    @Expose
    private Source source;
    private int source_id;
    @SerializedName("author")
    @Expose
    @ColumnInfo(name = "author")
    private String author;
    @SerializedName("title")
    @Expose
    @ColumnInfo(name = "title")
    private String title;
    @SerializedName("description")
    @Expose
    @ColumnInfo(name = "description")
    private String description;
    @SerializedName("url")
    @Expose
    @ColumnInfo(name = "url")
    private String url;
    @SerializedName("urlToImage")
    @Expose
    @ColumnInfo(name = "urlToImage")
    private String urlToImage;
    @SerializedName("publishedAt")
    @Expose
    @ColumnInfo(name = "publishedAt")
    private String publishedAt;
    @SerializedName("content")
    @Expose
    @ColumnInfo(name = "content")
    private String content;

    public int getSource_id() {
        return source_id;
    }

    public void setSource_id(int source_id) {
        this.source_id = source_id;
    }

    public Source getSource() {
        return source;
    }

    public void setSource(Source source) {
        this.source = source;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getUrlToImage() {
        return urlToImage;
    }

    public void setUrlToImage(String urlToImage) {
        this.urlToImage = urlToImage;
    }

    public String getPublishedAt() {
        return publishedAt;
    }

    public void setPublishedAt(String publishedAt) {
        this.publishedAt = publishedAt;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

}

File: app/src/main/java/edgar/yodgorbek/yangiliklar/model/Article.kt
package edgar.yodgorbek.yangiliklar.model

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.ForeignKey
import android.arch.persistence.room.PrimaryKey

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName


@Entity(tableName = "article", foreignKeys = [ForeignKey(entity = Source::class, parentColumns = ["id"], childColumns = ["source_id"])])
class Article {
    @PrimaryKey
    @SerializedName("source")
    @Expose
    lateinit var source: Source
    var source_id: Int = 0
    @SerializedName("author")
    @Expose
    @ColumnInfo(name = "author")
    var author: String? = null
    @SerializedName("title")
    @Expose
    @ColumnInfo(name = "title")
    var title: String? = null
    @SerializedName("description")
    @Expose
    @ColumnInfo(name = "description")
    var description: String? = null
    @SerializedName("url")
    @Expose
    @ColumnInfo(name = "url")
    var url: String? = null
    @SerializedName("urlToImage")
    @Expose
    @ColumnInfo(name = "urlToImage")
    var urlToImage: String? = null
    @SerializedName("publishedAt")
    @Expose
    @ColumnInfo(name = "publishedAt")
    var publishedAt: String? = null
    @SerializedName("content")
    @Expose
    @ColumnInfo(name = "content")
    var content: String? = null

}
